### ISSUE TITLE
Possible bug when calculating running loss

### ISSUE THREAD

**Issue by HenryDashwood** [keitakurita/torchtext_tutorials#8]
Hi there!

I've been looking at your training loop in the text classification tutorial and I see that you calculate a total running loss before averaging to get the average loss after the loop ends. 

Looking at this line though I'm a little confused
```running_loss += loss.data[0] * x.size(0)```
When I print out the shape of ```x``` I get these values
<img width="173" alt="Screen Shot 2019-05-08 at 1 29 50 PM" src="https://user-images.githubusercontent.com/17177967/57395190-7942d680-7195-11e9-8746-7303d00fbd1f.png">
The batch sizes appear to be at index 1 so should the running loss line read ```running_loss += loss.data[0] * x.size(1)```?

Entirely possible that I've made the mistake here but thought I would ask anyway!

**Comment by rxc205** [comment]
> Hi there!
> 
> I've been looking at your training loop in the text classification tutorial and I see that you calculate a total running loss before averaging to get the average loss after the loop ends.
> 
> Looking at this line though I'm a little confused
> `running_loss += loss.data[0] * x.size(0)`
> When I print out the shape of `x` I get these values
> <img alt="Screen Shot 2019-05-08 at 1 29 50 PM" width="173" src="https://user-images.githubusercontent.com/17177967/57395190-7942d680-7195-11e9-8746-7303d00fbd1f.png">
> The batch sizes appear to be at index 1 so should the running loss line read `running_loss += loss.data[0] * x.size(1)`?
> 
> Entirely possible that I've made the mistake here but thought I would ask anyway!

hello ,i have the same issue , have you ever solved this issue? just running_loss += loss.data[0] * x.size(1) is ok ?

**Comment by rxc205** [comment]
> Hi there!
> 
> I've been looking at your training loop in the text classification tutorial and I see that you calculate a total running loss before averaging to get the average loss after the loop ends.
> 
> Looking at this line though I'm a little confused
> `running_loss += loss.data[0] * x.size(0)`
> When I print out the shape of `x` I get these values
> <img alt="Screen Shot 2019-05-08 at 1 29 50 PM" width="173" src="https://user-images.githubusercontent.com/17177967/57395190-7942d680-7195-11e9-8746-7303d00fbd1f.png">
> The batch sizes appear to be at index 1 so should the running loss line read `running_loss += loss.data[0] * x.size(1)`?
> 
> Entirely possible that I've made the mistake here but thought I would ask anyway!
        running_loss+=loss.item() * x.size(0)， works！！！

**Comment by HenryDashwood** [comment]
I ended up using .size(1). It seems to work

**Comment by gitfourteen** [comment]
`x.size(0)` is the lengths of **comment_text** for the current batch, which have been automatically padded by **torchtext** in advance. The meaning of multiplying `x.size(0)` in **scalar loss function**  is not clear. Maybe the author consider it something like **batch weights** ? But still NO sense.
